crates/wash-lib/src/start/nats.rs

```diff
@@ -262,40 +266,62 @@ where
     P: AsRef<Path>,
     T: Into<Stdio>,
 {
-    let host_addr = format!("{}:{}", config.host, config.port);
-    // If we can connect to the local port, NATS won't be able to listen on that port
-    if tokio::net::TcpStream::connect(&host_addr).await.is_ok() {
-        bail!(
-            "could not start NATS server, a process is already listening on {}:{}",
-            config.host,
-            config.port
-        );
-    }
-    let child = if let Some(parent_path) = bin_path.as_ref().parent() {
-        let config_path = parent_path.join(NATS_SERVER_CONF);
-        let host = config.host.clone();
-        let port = config.port;
-        config.write_to_path(&config_path).await?;
-        Command::new(bin_path.as_ref())
+    if let Some(natscfg_path) = config.config_path {
+        // Use the existing config path
+        let child = Command::new(bin_path.as_ref())
```

Comment:
I think we can get away with checking for this later -- what do you think? 

This is a NIT but I think we can start building the `Command` mutably (i.e. `let mut child = Command::new(...)`), add the shared stuff, *then* check for whether a NATS config path was provided and return early succinctly!

Something like:

```rust
let mut child = Command::new(bin_path.as_ref())
.stderr(stderr)
.stdin(Stdio::null())
.arg("-js")
.arg("--config");

if let Some(nats_cfg_path) = config.config_path {
child = child.arg(natscfg_path.join(NATS_SERVER_CONF))
.arg("--pid")
.arg(nats_cfg_path.join("NATS_SERVER_PID")
.spawn()
......
}

// add the rest

```

This also makes me think -- if we're expecting the NATS config path to be a directory, can we make this explicit in the name of the argument? We should also maybe check that the path is a directory.

I know that when I see `path` I think file -- so I might think that `--nats-config` should be the path to a NATS config file, not the directory it's in. 

Something like `--nats-config-dir`/`--nats-dir`? Because we actually *write* the PID file into that directory too. It's not like we can pass a readonly directory mount in.

Reply:
So I'll refactor `--nats-config` to `--nats-config-dir` and add a `is_dir()` check as you suggested.
I'll make `child` a mutable too

Reply:
```rust
let mut child = Command::new(bin_path.as_ref())
.stderr(stderr)
.stdin(Stdio::null())
.arg("-js")
.arg("--config");
```
The temporary value of child gets freed at the end of the statement, so the Command::new(bin_path.as_ref()) does not persist. It needs `.spawn()` attached to the command call.
I initially planned on taking this approach :point_down:  , however `wait_for_server` has a dependency on `host_addr` which will have conflicts with the host and port specified in the nats.conf file, so I think it would be best if we checked for `nats_cfg_path` in advance

```rust
async fn start_nats_server_2<P, T>(bin_path: P,config: NatsConfig,stderr: T,) -> Result<tokio::process::Child>
...
....
//rest of the code
.....
.......
    let bin_path_ref = bin_path.as_ref();

    if let Some(parent_path) = bin_path_ref.parent() {
        let config_path = parent_path.join(NATS_SERVER_CONF);
        let host = config.host.clone();
        let port = config.port;

        let mut cmd_args = vec!["-js".to_string(), "--config".to_string()];

        if let Some(nats_cfg_path) = &config.config_path {
            if nats_cfg_path.is_dir(){
                cmd_args.push(nats_cfg_path.join(NATS_SERVER_CONF).to_string_lossy().to_string());
                cmd_args.push("--pid".to_string());
                cmd_args.push(nats_cfg_path.join(NATS_SERVER_PID).to_string_lossy().to_string());
            }else{
                bail!("couldn't find directory path");
            }
        }  else {
            config.write_to_path(&config_path).await?;
            cmd_args.push(config_path.to_string_lossy().to_string());
            cmd_args.push("--addr".to_string());
            cmd_args.push(host);
            cmd_args.push("--port".to_string());
            cmd_args.push(port.to_string());
        }

        let child = Command::new(bin_path_ref)
            .stderr(stderr.into())
            .stdin(Stdio::null())
            .args(&cmd_args) //additional arguements get added here
            .spawn()
            .map_err(anyhow::Error::from)?;

        wait_for_server(&host_addr, "NATS server")
            .await
            .map(|()| child)
    }
....
....
    }
}
```